config/routes.rb

```diff
@@ -8,4 +12,16 @@
       resources :likes, only: [:create]
     end
   end
+  
+  # API ROUTES
+  namespace :api do # /api
+    namespace :v1 do # /api/v1
+      resources :users, only: [:index,:show] do # api/v1/users
+        resources :posts, only: [:index,:show], format: :json do # api/v1/users/:user_id/posts
+          resources :comments, format: :json # api/v1/users/:user_id/posts/:post_id/comments
+        end
+      end
+    end
+  end
+  post "/login", to: "authors#login"
```

Comment:
- [ ] Nice working on your authentication, kindly check I can't seem to register or login I get the below error in the screenshot when I try.
![image](https://user-images.githubusercontent.com/48371156/202828358-44efc0e1-809d-4ec6-82dd-62fd78c93790.png)

Reply:
Hi @danielufeli :wave: 

I used a gem called [doorkeeper](https://github.com/doorkeeper-gem/doorkeeper),

So you have to create an OAuth Application running this in rails console: `Doorkeeper::Application.create(name: "Android client", redirect_uri: "", scopes: "")` and to log in you need to follow [this steps](https://rubyyagi.com/rails-api-authentication-devise-doorkeeper/#setup-doorkeeper-gem:~:text=URL%2DEncoded%20form.-,Login,with%20grant_type%2C%20email%2C%20password%2C%20client_id%20and%20client_secret%20attributes.,-As%20we%20are).
**when you log in, you have to copy the access token**

The way I created the API, to add a comment you need to add `/api/v1` right after the 3000 in the url (as in the image below)
![Screenshot from 2022-11-18 23-02-09](https://user-images.githubusercontent.com/99205303/202829334-ea1feb31-4d63-42ce-be79-27abd8ce1d61.png)

and need to use the **access token you copied** (type Bearer) on the Authorization tab (image below)
![Screenshot from 2022-11-18 23-09-07](https://user-images.githubusercontent.com/99205303/202829415-4c165302-2e2d-4b4e-beef-6acf58ba24ba.png)